assets/javascripts/discourse/components/chat-composer-uploads.js

```diff
@@ -0,0 +1,172 @@
+import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import UppyMediaOptimization from "discourse/lib/uppy-media-optimization-plugin";
+import ComposerUploadUppy from "discourse/mixins/composer-upload-uppy";
+import {
+  authorizedExtensions,
+  authorizesAllExtensions,
+} from "discourse/lib/uploads";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+
+export default Component.extend(ComposerUploadUppy, {
+  classNames: ["chat-composer-uploads"],
+
+  // represents the component state of uploads
+  _uploads: null,
+  inProgressUploads: null,
+  uploads: null,
+  uploadProcessorActions: null,
+  uploadPreProcessors: null,
+  uploadMarkdownResolvers: null,
```

Comment:
I think I can make a core change to just ignore this property if it's null, we don't need it for chat, it's only used if you need to mess with the upload markdown, and since we aren't even using that markdown in chat we can just throw it away. I can make a companion core PR perhaps that we can use to clean up after this one.

Reply:
Leaving this here but nevermind it, see the main PR comment.